packages/devtools_app/lib/src/screens/vm_developer/object_inspector/vm_instance_display.dart

```diff
@@ -0,0 +1,244 @@
+              ),
+              Flexible(
+                child: FutureBuilder(
+                  future: _initialized,
+                  builder: (context, snapshot) {
+                    if (snapshot.connectionState != ConnectionState.done)
+                      return const CenteredCircularProgressIndicator();
+                    return ExpandableVariable(
+                      variable: _root,
+                      dataDisplayProvider: (variable, onPressed) {
+                        return DisplayProvider(
+                          controller: widget.controller,
+                          variable: variable,
+                          onTap: onPressed,
+                        );
+                      },
+                    );
+                  },
+                ),
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _InstanceViewer extends StatelessWidget {
+  const _InstanceViewer({
+    required this.controller,
+    required this.instance,
+  });
+
+  final ObjectInspectorViewController controller;
+  final InstanceObject instance;
+
+  @override
+  Widget build(BuildContext context) {
+    return VmObjectDisplayBasicLayout(
+      controller: controller,
+      object: instance,
+      generalDataRows: [
+        serviceObjectLinkBuilderMapEntry<ClassRef>(
+          controller: controller,
+          key: 'Object Class',
+          object: instance.obj.classRef!,
+        ),
+        selectableTextBuilderMapEntry(
+          'Shallow Size',
+          prettyPrintBytes(
+            instance.obj.size ?? 0,
+            includeUnit: true,
+            kbFractionDigits: 1,
+            maxBytes: 512,
+          ),
+        ),
+        MapEntry(
+          'Reachable Size',
+          (context) => RequestableSizeWidget(
+            fetching: instance.fetchingReachableSize,
+            sizeProvider: () => instance.reachableSize,
+            requestFunction: instance.requestReachableSize,
+          ),
+        ),
+        MapEntry(
+          'Retained Size',
+          (context) => RequestableSizeWidget(
+            fetching: instance.fetchingRetainedSize,
+            sizeProvider: () => instance.retainedSize,
+            requestFunction: instance.requestRetainedSize,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class DisplayProvider extends StatelessWidget {
+  const DisplayProvider({
+    super.key,
+    required this.variable,
+    required this.onTap,
+    required this.controller,
+  });
+
+  final DartObjectNode variable;
+  final VoidCallback onTap;
+  final ObjectInspectorViewController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    // TODO(devoncarew): Here, we want to wait until the tooltip wants to show,
+    // then call toString() on variable and render the result in a tooltip. We
+    // should also include the type of the value in the tooltip if the variable
+    // is not null.
```

Comment:
was this TODO accidentally copied from somewhere? I don't think Devon will be jumping in to play with tooltips any time soon 😆